features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/tracking/ActivityLifecycleTrackingStrategy.kt

```diff
@@ -92,16 +91,11 @@ abstract class ActivityLifecycleTrackingStrategy :
 
     @MainThread
     override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
-        val intent = activity.intent
-        val extras = intent.safeExtras
-        val testId = extras?.getString("_dd.synthetics.test_id")
-        val resultId = extras?.getString("_dd.synthetics.result_id")
-        if (!testId.isNullOrBlank() && !resultId.isNullOrBlank()) {
-            (GlobalRumMonitor.get(sdkCore) as? AdvancedRumMonitor)
-                ?.setSyntheticsAttribute(
-                    testId,
-                    resultId
-                )
+        if (::sdkCore.isInitialized) {
```

Comment:
While there is nothing wrong with this condition the way I see it now, I wonder which test was failing without adding it ? What happened before if the `sdkCore` was not initialized ? Was it defaulting to a `NoOp` ?

Reply:
Previously we checked if the intent contained the synthetics attributes in the extras, and only then we would use the `sdkCore`. 

With this PR we delegate the responsibility of checking the intent extras to the new RUMMonitor method, so we have to access the core before checking the intent extras. 

The problem occurs because - in that specific test - the register method is never called.

This is the test: [when activity created will not delegate to the bundled strategies() ](https://github.com/DataDog/dd-sdk-android/blob/6319509dfae068ade55741b6b57173670f0f22cf/features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/MixedViewTrackingStrategyTest.kt#L84)